src/cogent3/app/composable.py

```diff
@@ -1023,15 +1012,16 @@ def wrapped(klass):
             klass.__slots__ += tuple(slot_attrs)
 
         # register this app
-        __app_registry[get_object_provenance(klass)] = klass.__name__
+        __app_registry[get_object_provenance(klass)] = composable
+
         return klass
 
     return wrapped(klass) if klass else wrapped
 
 
 def is_composable(obj):
     """checks whether obj has been registered by the composable decorator"""
-    return get_object_provenance(obj) in __app_registry
+    return __app_registry.get(get_object_provenance(obj)) or False
```

Comment:
I changed this function, because we have a class, hypothesis. This class has a base class which is composable app. So, I changed this function to find inherited composable apps.

Reply:
Good point. The problem is by not using the decorator we don't get app registration. See my comment on the `hypothesis` code.